test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public class AspirationalExamples
+    {
+        public class IrisPrediction
+        {
+            public string PredictedLabel;
+        }
+
+        public class IrisExample
+        {
+            public float[] Features;
+        }
+
+        public void FirstExperienceWithML()
+        {
+            // This is the 'getting started with ML' example, how we see it in our new API.
+            // It currently doesn't compile, let alone work, but we still can discuss and improve the syntax.
+
+            // Initialize the environment.
+            using (var env = new TlcEnvironment())
+            {
+                // Load the data into the system.
+                string dataPath = "iris-data.txt";
+                var data = TextReader.FitAndRead(env, dataPath, c => (Label: c.LoadString(0), Features: c.LoadFloat(1, 4)));
```

Comment:
Name will need to be something else so that it doesn't collide with System.IO.TextReader, TextDataReader or something like that.  Also `ReadAndFit` seems more natural to me.

What sort of object is `c`?  an you point me to the code methods that do `Load*` so I can understand this better?

Reply:
TextDataReader sounds OK.

'ReadAndFit': well, the code as written is actually a shorthand for
`.Fit(dataPath).Read(dataPath)`, so ReadAndFit would be incorrect

---
In reply to: [207608639](https://github.com/dotnet/machinelearning/pull/639#discussion_r207608639) [](ancestors = 207608639)